src/Umbraco.Web/HealthCheck/NotificationMethods/SlackNotificationMethod.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Slack.Webhooks;
+using Umbraco.Core.Configuration.HealthChecks;
+
+namespace Umbraco.Web.HealthCheck.NotificationMethods
+{
+    [HealthCheckNotificationMethod("slack")]
+    public class SlackNotificationMethod : NotificationMethodBase, IHealthCheckNotificatationMethod
+    {
+        public SlackNotificationMethod(bool enabled, bool failureOnly, HealthCheckNotificationVerbosity verbosity,
+                string webHookUrl, string channel, string username)
+            : base(enabled, failureOnly, verbosity)
+        {
+            WebHookUrl = webHookUrl;
+            Channel = channel;
+            Username = username;
+        }
+
+        public string WebHookUrl { get; set; }
```

Comment:
Similar to the above, if these are get/set they are mutable after the object is created. without looking at all of the code i'm unsure what the ramifications are of that. If this is a singleton then you have mutable properties over the lifetime of the app which isn't good.